src/rez/utils/data_utils.py

```diff
@@ -14,13 +13,29 @@ class SourceCode(object):
     def __init__(self, source):
         self.source = source.rstrip()
 
+    def maybe_indent(self):
+        if self.source and self.source[0] in (' ', '\t'):
+            # outer indent(s) follow, perhaps comments
+            self.source = "if True:\n" + self.source
```

Comment:
Probably not a good idea to change in place - SourceCode instances are stored in Packages/Variants, which are stored in a cache. Better to return a copy. This could cause bugs in threaded code.